midiproc/buffer.c

```diff
@@ -0,0 +1,253 @@
+//
+// Copyright(C) 2017 Alex Mayfield
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//     A simple buffer and reader implementation.
```

Comment:
I wonder if this file is necessary. I understand the motivation, but in this particular situation, it's reasonable to assume you'll never use a `chocolate-doom.exe` with a `midiproc.exe` that aren't built from the same version and same compiler. So maybe it's simplest to just put stuff in a struct and write the contents to the stream?

Normally I avoid doing this because of the usual reasons of different compilers and platforms doing struct packing differently. Feel free to use the struct packing #defines if that helps.